telethon/telegram_bare_client.py

```diff
@@ -300,15 +313,15 @@ def _get_dc(self, dc_id, ipv6=False, cdn=False):
 
             return next(
                 dc for dc in TelegramBareClient._config.dc_options
-                if dc.id == dc_id and bool(dc.ipv6) == ipv6 and bool(dc.cdn) == cdn
+                if dc.id == dc_id and bool(dc.ipv6) == self._use_ipv6 and bool(dc.cdn) == cdn
```

Comment:
Here again, I noticed you use `'[2001:67c:4e8:f002::a]'` as the IPv6 string (notice the square braces `'[]'`). Why are these needed? Telegram's `dc.ipv6` don't have these braces. Will it still work?

Reply:
OK, I will test it today/tomorrow and let you know.

Reply:
Easiest way is to manually set a different datacenter on the session file so when you connect the library will need to change to your closest one.

Reply:
Well, as I've just tested, in case of ipv6 connection, the server address MUST be in `'[]'` and the port MUST be 80, or server/proxy will return an Bad Gateway or Timeout error.